src/widgets/Survey/ui/ScreenManager.tsx

```diff
@@ -23,6 +26,39 @@ export const ScreenManager = ({ openTimesUpModal }: Props) => {
     ),
   );
 
+  const { saveAutoCompletion } = AutoCompletionModel.useAutoCompletionStateManager();
+
+  const onEntityTimerFinish = useCallback(() => {
+    const activitiesToSubmit: string[] =
+      CompletionContructService.extractActivityIdsToSubmitByParams({
```

Comment:
Please use a separate function class for this purpose

Reply:
Done